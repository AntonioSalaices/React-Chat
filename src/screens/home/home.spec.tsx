import { render, RenderResult, fireEvent, act } from '@testing-library/react';
import Home from './home';

describe('<Home />:: Spec', () => {
  let component: RenderResult;

  beforeEach(() => {
    component = render(<Home />);
  });
  test('should render correctly', () => {
    const { container } = component;

    expect(container).toMatchSnapshot();
  });

  test('should search something and execute debounce function', async () => {
    const { getByTestId } = component;
    jest.useFakeTimers();
    await act(async () => {
      fireEvent.change(getByTestId('searchInput'), { target: { value: 'testing' } });
      jest.advanceTimersByTime(300);
    });
  });
});
