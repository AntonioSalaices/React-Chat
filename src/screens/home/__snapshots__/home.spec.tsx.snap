// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home />:: Spec should render correctly 1`] = `
<div>
  <div
    class="container"
    id="main-container"
  >
    <div
      class="col justify-center align-center"
    >
      <div
        class="col-11-xs col-11-sm col-5-md col-5-xl"
      >
        <div
          class="gap-2"
        >
          <label>
            [missing "en-US.form.inputs.search" translation]
            <input
              class="input-primary"
              data-testid="searchInput"
              placeholder="[missing "en-US.form.inputs.search" translation]"
              type="text"
              value=""
            />
          </label>
          <p
            class="text-primary-light-7"
          >
            [missing "en-US.subPlaceholder.search" translation]
          </p>
        </div>
      </div>
      <div
        class="col-11-xs col-11-sm col-5-md col-5-xl"
      >
        <div
          class="gap-2"
        >
          <label>
            <input
              class="input-primary"
              data-testid="resizeInput"
              max="100"
              min="50"
              type="range"
              value=""
            />
          </label>
          <p
            class="text-primary-light-7"
          >
            [missing "en-US.form.inputs.fontSize" translation]
          </p>
        </div>
      </div>
    </div>
    <div
      class="container"
      id="main-container"
    >
      <div
        class="row justify-center"
      >
        <span
          style="display: inline-block; position: relative; width: 50px; height: 50px; vertical-align: middle; font-size: 0px; animation: __react-md-spinner-animation__root-rotate 1568.235294117647ms linear infinite;"
        >
          <span
            style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-1-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 5px; border-top-style: solid;"
            />
          </span>
          <span
            style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-2-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 5px; border-top-style: solid;"
            />
          </span>
          <span
            style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-3-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 5px; border-top-style: solid;"
            />
          </span>
          <span
            style="box-sizing: border-box; display: block; position: absolute; width: 100%; height: 100%; border-color: #5c217f; white-space: nowrap; opacity: 1; animation-name: __react-md-spinner-animation__fill-unfill-rotate, __react-md-spinner-animation__layer-4-fade-in-out; animation-duration: 5332ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite;"
          >
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: 0px; transform: rotate(129deg); animation-name: __react-md-spinner-animation__left-spin; border-color: #5c217f transparent transparent #5c217f;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: relative; width: 50%; height: 100%; overflow: hidden;"
            >
              <span
                style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; bottom: 0px; width: 200%; border-width: 5px; border-style: solid; animation-duration: 1333ms; animation-timing-function: cubic-bezier(.4, 0, .2, 1); animation-iteration-count: infinite; left: -100%; transform: rotate(-129deg); animation-name: __react-md-spinner-animation__right-spin; border-color: #5c217f #5c217f transparent transparent;"
              />
            </span>
            <span
              style="display: inline-block; box-sizing: border-box; position: absolute; top: 0px; border-radius: 50%; left: 45%; width: 10%; border-width: 5px; border-top-style: solid;"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
